Fix network diagram stories, change JSON to JS objects for StackBlitz examples, bump deps within semver, make prettier and eslint behave better

nstuyvesant opened the pull request:
### Updates
- Wrong CSS class was being applied to react ShapeNode
- Story for React diagrams was missing height and width on foreignObject (updated to match Angular diagram story)
- Story for React diagrams - size was missing on icon (updated to match Angular diagram story)
- `@carbon/icons-angular` and `@carbon/icons-react` have slightly different rendering behavior. The latter adds `fill="currentColor"` if a fill color is not specified. Added a specific fill of black in story for React to match look of Angular story.
- StackBlitz examples now have data and options as JavaScript or TypeScript instead of JSON so functions like `valueFormatter` can be added to the examples.
- Fix story backgrounds for diagrams to be white and correct way backgrounds were originally set (ran once globally vs. per story)
- Bumps dependencies within semver range (storybook, vite, svelte, sass, node, yarn, etc.)
- Fixes https://github.com/carbon-design-system/carbon-charts/issues/1613
- eslint config was missing for root-level
- Updates prettier config to ESM
- Replaced deprecated `@carbon/icons-angular` with `@carbon/icons` per the [README.md](https://github.com/carbon-design-system/carbon-icons-angular/blob/master/README.md)
- Simplifies scripts (eliminates 3 bash shell scripts in favor of package.json scripts)
- Updates keywords for packages to improve discoverability on npmjs

<img width="490" alt="image" src="https://github.com/carbon-design-system/carbon-charts/assets/5770938/b503bc79-11ca-4669-a777-a86bad8e02af">

nstuyvesant commented on packages/core/src/components/essentials/tooltip-axis.ts:
This what fixes issue 1631

nstuyvesant commented on packages/core/src/demo/utils/stackblitz/react.ts:
```diff
@@ -66,7 +65,10 @@ class App extends React.Component {
     ></ ${demo.chartType.vanilla}>
   )
 }
-ReactDOM.render(<App />, document.getElementById('root'))
```
This was forcing React 17 behavior instead of 18.

nstuyvesant commented on packages/react/src/diagrams/ShapeNode.tsx:
```diff
@@ -40,7 +40,7 @@ const ShapeNode: React.FC<
 	const circleClasses = classnames(namespace, {
 		[`${namespace}--stacked`]: stacked,
 		[`${namespace}--${shape}`]: shape,
-		[`${namespace}--${tag}`]: tag,
```
@theiliad - one of the two problems you identified

nstuyvesant commented on .yarnrc.yml:
```diff
@@ -1,19 +1,7 @@
 
-yarnPath: .yarn/releases/yarn-3.5.1.cjs
-
-# Required to prevent "Error: Cannot find module '@storybook/react-webpack5/preset'"
```
This was there due to a bug in Storybook that was fixed in 7.1.1 so now removed.

nstuyvesant commented on .yarnrc.yml:
```diff
@@ -1,19 +1,7 @@
-nmHoistingLimits: workspaces
```
This was a workaround for a Storybook issue that has since been resolved. Removing this setting causes most packages to hoist unless there's a specific version difference. This also solved the plugin-prettier-svelte issue with lerna.

nstuyvesant commented on .eslintrc.json:
```diff
@@ -0,0 +1,6 @@
+    "ecmaVersion": 2023,
+    "sourceType": "module"
+  }
+}
```
For some reason, we never had an eslint config file at the top level of the repo. This is needed for scripts/update-angular-dependency-version.mjs and prettier.config.mjs.

nstuyvesant commented on packages/angular/src/stories/all.stories.ts:
```diff
@@ -43,7 +43,7 @@ const getTemplate = (componentSelector: string) => `
 		</h3>
 		<p class="props">
-		  <b>Props:</b> data, <a href="https://carbon-design-system.github.io/carbon-charts/documentation/modules/interfaces.html" target="_blank">options</a>
+		  <b>Props:</b>&nbsp;<a href="/?path=/docs/docs-tutorials-tabular-data-format--docs">data</a>, <a href="https://carbon-design-system.github.io/carbon-charts/documentation/modules/interfaces.html" target="_blank">options</a>
```
Needed link to tabular data format MDX

nstuyvesant commented on packages/angular/src/stories/all.stories.ts:
```diff
@@ -64,10 +64,14 @@ storybookDemoGroups.forEach((demoGroup: DemoGroup) => {
 	// Create stories within story group
 	demoGroup.demos.forEach((demo: Demo) => {
 		demo.options.theme = DEFAULT_THEME
-		document.documentElement.setAttribute('data-carbon-theme', DEFAULT_THEME)
```
Bug because the theme was set once when storybook starts rather than each time an individual story loads. If you have some stories with the white theme and others g100, it doesn't work well this way.

nstuyvesant commented on packages/angular/src/stories/diagrams/0-diagrams-start-here.stories.ts:
```diff
@@ -1,20 +1,30 @@
 import { storiesOf, moduleMetadata } from '@storybook/angular'
-import { buildElbowPathString, type Coordinates } from '@carbon/charts'
-import { UserModule, WikisModule } from '@carbon/icons-angular'
```
The package `@carbon/icons-angular` was recently deprecated per its read me. I switched it over to using `@carbon/icons` directly (ESM).

nstuyvesant commented on packages/angular/src/stories/diagrams/carbon-icons.d.ts:
```diff
@@ -0,0 +1,23 @@
+
+	const icon: CarbonIcon
+	export default icon
+}
```
Not sure why but `@carbon/icons` has no types.

nstuyvesant commented on packages/angular/src/stories/diagrams/diagrams-nodes-card.stories.ts:
```diff
@@ -1,6 +1,16 @@
+import screenIcon from '@carbon/icons/es/screen/16'
+import chevronDownIcon from '@carbon/icons/es/chevron--down/16'
 import { CardNodeModule } from '../../'
-import { ScreenModule, ChevronDownModule } from '@carbon/icons-angular'
```
Again, replacing `@carbon/icons-angular` with `@carbon/icons`

nstuyvesant commented on packages/angular/src/stories/diagrams/utils.ts:
```diff
@@ -1,5 +1,5 @@
 export const getTemplate = (content: string) => `
-<div class="container theme--white">
+<div class="container theme--white" style="min-height:140px;">
```
For certain shapes, the height of the background was half the height of the shape.

nstuyvesant commented on packages/core/scss/demos.scss:
```diff
@@ -36,6 +36,10 @@ $css--reset: false; // Do not include reset CSS
 
 html {
+	@include styles.theme(styles.$g100);
+}
```
The default theme for stories is now G100 - not white.

nstuyvesant commented on prettier.config.mjs:
```diff
@@ -1,10 +1,14 @@
 	singleQuote: true,
 	printWidth: 100,
 	trailingComma: 'none',
+	useTabs: true,
```
Or else prettier will want to update 500+ files.

nstuyvesant commented on .yarnrc.yml:
```diff
@@ -1,19 +1,8 @@
-#       "@storybook/angular": "*"
-#       "@storybook/react-webpack5": "*"
-#       "react": "*"
-#       "react-dom": "*"
```
Commented lines were to address a Storybook bug (now fixed)

nstuyvesant commented on package.json:
```diff
@@ -1,16 +1,17 @@
+    "pub:demos": "lerna run pub:demo",
+    "build": "yarn build:packages && yarn build:demos && yarn clean:pages && yarn pub:demos",
     "deploy": "bash scripts/deploy.sh",
-    "deploy:next": "bash scripts/next.sh",
```
This was a temporary script - no longer needed

nstuyvesant commented on packages/core/src/components/essentials/tooltip-axis.ts:
```diff
@@ -9,8 +9,8 @@ export class AxisChartsTooltip extends Tooltip {
-		const data = e.detail.data
-		if (!data.length || !data[0]) {
+		const { data } = e.detail
+		if (!data || !data.length || !data[0]) {
```
Fix for https://github.com/carbon-design-system/carbon-charts/issues/1613

theiliad replied:
we should have this kind of thing in a separate PR, but nbd for now

nstuyvesant commented on packages/core/src/demo/utils/stackblitz/angular.ts:
```diff
@@ -1,30 +1,28 @@
+		'@angular/common': version.angular,
+		'@angular/compiler': version.angular,
+		'@angular/core': version.angular,
+		'@angular/platform-browser': version.angular,
```
prettier replacing spaces with tabs :-(

nstuyvesant commented on packages/core/src/demo/utils/stackblitz/angular.ts:
```diff
@@ -65,20 +62,18 @@ import dataJson from './data.json'
 
-  const stylesCss =
-`@import '@carbon/styles/css/styles.css';
+	const stylesCss = `@import '@carbon/styles/css/styles.css';
```
prettier reformatting only

nstuyvesant commented on packages/core/src/demo/utils/stackblitz/react.ts:
```diff
@@ -66,7 +65,10 @@ class App extends React.Component {
+const root = ReactDOM.createRoot(document.getElementById('root'))
+root.render(
+  <App />
+)
```
This makes the StackBlitz React examples use React 18 (createRoot) behavior.

theiliad replied:
yes I was going to do this!

nstuyvesant commented on packages/vue/src/stories/all.stories.ts:
```diff
@@ -70,7 +74,7 @@ storybookDemoGroups.forEach((demoGroup: DemoGroup) => {
 
 						<p class="props">
-							<b>Props:</b> <a href="/?path=/docs/docs-tutorials-tabular-data-format">data</a>, <a href="https://carbon-design-system.github.io/carbon-charts/documentation/modules/interfaces.html" target="_blank">options</a>
+							<b>Props:</b>&nbsp;<a href="/?path=/docs/docs-tutorials-tabular-data-format">data</a>, <a href="https://carbon-design-system.github.io/carbon-charts/documentation/modules/interfaces.html" target="_blank">options</a>
```
Needed a slightly bigger space between colon and `data` hyperlink

nstuyvesant commented on scripts/build-demos.sh:
Replaced by "build" script in package.json...
```json
    "build": "yarn build:packages && yarn build:demos && yarn clean:pages && yarn pub:demos"
```

nstuyvesant commented on scripts/build-packages-and-demos.sh:
Replaced by "build" script in package.json...
```json
    "build": "yarn build:packages && yarn build:demos && yarn clean:pages && yarn pub:demos"
```

nstuyvesant commented on scripts/copy-demos-to-deploy-dir.sh:
Replaced by "pub:demos" script in package.json
```json
    "pub:demos": "lerna run pub:demo"
```
Calls "pub:demo" script in each package allowing for more granularity.

nstuyvesant commented on scripts/next.sh:
Script no longer needed

theiliad commented on packages/core/src/demo/utils/stackblitz/object-to-string.ts:
```diff
@@ -0,0 +1,44 @@
+		str += padding + ']'
+	} else if (typeof obj === 'object' && obj !== null) {
+		if (obj instanceof Date) {
+			str += `'${obj.toISOString()}'`
```
can we make this shoot out a `new Date(...)` instead of a string?

nstuyvesant replied:
Done

theiliad commented on packages/core/.storybook/manager-head.html:
```diff
@@ -7,7 +7,7 @@
 
 <!-- Open Graph -->
-<meta property="og:title" content="Carbon Charts - JavaScript (vanilla)" />
+<meta property="og:title" content="Carbon Charts - Vanilla JavaScript" />
```
```suggestion
<meta property="og:title" content="Carbon Charts - vanilla JavaScript" />
```

theiliad commented on packages/core/.storybook/manager-head.html:
```diff
@@ -20,6 +20,6 @@
 <!-- Social -->
 <meta name="twitter:card" content="summary_large_image" />
-<meta name="twitter:image:alt" content="Carbon Charts - JavaScript (vanilla)" />
+<meta name="twitter:image:alt" content="Carbon Charts - Vanilla JavaScript" />
```
```suggestion
<meta name="twitter:image:alt" content="Carbon Charts - vanilla JavaScript" />
```

theiliad reviewed (approved):
LGTM

theiliad reviewed (changes requested):
Some of the stackblitz examples are now broken

1. https://deploy-preview-1631--carbon-charts-core.netlify.app/?path=/story/utility-toolbar--vertical-stacked-bar-time-series

<img width="739" alt="image" src="https://github.com/carbon-design-system/carbon-charts/assets/14989804/162c4de6-9404-4eeb-982e-368236653a4c">

2. https://deploy-preview-1631--carbon-charts-core.netlify.app/?path=/story/utility-locale--turkish-locale-date-fns-locales

<img width="786" alt="image" src="https://github.com/carbon-design-system/carbon-charts/assets/14989804/617601bd-57c7-4c63-bea3-f1455c6d932d">

3. https://deploy-preview-1631--carbon-charts-core.netlify.app/?path=/story/complex-charts-choropleth-experimental--geo-data

![image](https://github.com/carbon-design-system/carbon-charts/assets/14989804/9d6c2836-3d8e-4d51-8870-431c6f36e4bd)

nstuyvesant commented on packages/core/src/demo/charts/line.ts:
```diff
@@ -273,7 +271,7 @@ export const lineTimeSeriesWithThresholdsOptions = {
 					value: new Date(2019, 0, 11),
 					label: 'Custom formatter',
-					valueFormatter: timeFormat('%b %d')
+					valueFormatter: e => new Intl.DateTimeFormat('en-US', { month: 'short', day: '2-digit' }).format(e)
```
Use standards approach to formatting.

nstuyvesant commented:
> Some of the stackblitz examples are now broken
> 
> 1. https://deploy-preview-1631--carbon-charts-core.netlify.app/?path=/story/utility-toolbar--vertical-stacked-bar-time-series
> 
> <img alt="image" width="739" src="https://user-images.githubusercontent.com/14989804/256713511-162c4de6-9404-4eeb-982e-368236653a4c.png">
> 2. https://deploy-preview-1631--carbon-charts-core.netlify.app/?path=/story/utility-locale--turkish-locale-date-fns-locales
> 
> <img alt="image" width="786" src="https://user-images.githubusercontent.com/14989804/256713588-617601bd-57c7-4c63-bea3-f1455c6d932d.png">
> 3. https://deploy-preview-1631--carbon-charts-core.netlify.app/?path=/story/complex-charts-choropleth-experimental--geo-data
> 
> ![image](https://user-images.githubusercontent.com/14989804/256713722-9d6c2836-3d8e-4d51-8870-431c6f36e4bd.png)

Fixed with latest push. I renamed those demos slightly. The update uses standards approach to locales and addresses ticks for both axes and the tooltips.

theiliad reviewed (approved):
LGTM
